### PR TITLE
Update title separator. Closes #121

### LINKED ISSUE #121: Allow customization of the title-separator
Hey guys
Currently the `{{ page.title }}`-Property and the `{{ site.title }}`-Property inside the `<title>`-Tag are allways separated with an `-`-Character.
Could you implement a way to customize the separator to use?
I'd, for example, like to use an `·` instead of an `-`

Best regards an' thanks in advance

### PR DESCRIPTION
**Pull request by jhabdas**
Replaces the "-" used as a title separator with "|".
Ref: https://moz.com/learn/seo/title-tag

Also updates docs to <s>specify title optional and fix errata</s> provide additional info.

### CODE REVIEW

**Comment by jhabdas** [comment]
Will get those style snafus cleaned up as I incorporate any feedback on the Pull. Thank you kindly.

File: lib/template.html [hunk 1]
```diff
@@ -5,21 +5,13 @@
...
-  {% if seo_site_title %}
-    {% assign seo_title = seo_title | append:" - " | append: seo_site_title %}
-  {% endif %}
-{% elsif seo_site_title %}
-  {% assign seo_title = seo_site_title %}
-  {% assign seo_page_title = seo_site_title %}
-
-  {% if site.description %}
-    {% assign seo_title = seo_title | append:" - " | append: site.description %}
-  {% endif %}
```
Comment (benbalter) [inline lib/template.html]: Can you explain what's going on here?

File: README.md [hunk 2]
```diff
@@ -6,10 +6,10 @@ A Jekyll plugin to add metadata tags for search engines and social networks to b
 
 ## What it does
 
-Jekyll SEO Tag adds the following meta tags to your site:
+Jekyll SEO Tag adds the following tags to your site:
 
-* Pages title (with site title appended when available)
-* Page description
+* Optional `title` delimited with a pipe
+* Meta description
```
Comment (pathawks) [inline README.md]: There are some things going on here that are outside the scope of this PR.

Meta in this sense is only meaningful to HTML developers. I suspect “Page Description” is more clear to most users.

File: spec/jekyll_seo_tag_spec.rb [hunk 3]
```diff
@@ -36,21 +36,43 @@
   end
 
   context "with page.title" do
-    let(:page) { make_page("title" => "foo") }
+    let(:page) { make_page("title" => "Page Title") }
```
Comment (pathawks) [inline spec/jekyll_seo_tag_spec.rb]: I'm not a fan of changing behavior and tests together in the same commit.

File: lib/template.html (hunk 1 above)
Comment (jhabdas) [inline lib/template.html]: Yes, I refactored out 8 lines of code, and covered them all with tests.

File: README.md (hunk 2 above)
Comment (jhabdas) [inline README.md]: SGTM. Will update and remove jargon.

File: spec/jekyll_seo_tag_spec.rb (hunk 3 above)
Comment (jhabdas) [inline spec/jekyll_seo_tag_spec.rb]: Will pull from PR

**Comment by jhabdas** [comment]
Review feedback incorporated, offenses resolved.

File: lib/template.html [hunk 4]
```diff
@@ -5,21 +5,13 @@
...
+{% if page.title and seo_site_title %}
+  {% assign seo_title = seo_title | append:" | " | append: seo_site_title %}
+{% elsif site.description %}
+  {% assign seo_title = seo_title | append:" | " | append: site.description %}
```
Comment (pathawks) [inline lib/template.html]: I don't think this does the same thing as what we originally intended.

The goal was for the title to be `page.title -site.title` or `site.title - site.description`

The only time `site.description` is in the title is when `page.title` is empty.

If `page.title` is not empty and `site.title` is empty, the title should just be `page.title`

Right?

Comment (jhabdas) [inline lib/template.html]: I added a negative test assertion on L70 to ensure this was always the case, dropping the AND check here as it wasn't doing anything after the refactor. This was done as a trade-off to improve readability over the previous branch logic removed while slightly obfuscating functionality now covered by tests.

Comment (pathawks) [inline lib/template.html]: That does not appear to be what L70 is testing.

I am talking about the case where `page.title` is set, `site.name` is not set, `site.description` is set.

In this case, the first `if` condition is false because there is no `seo_site_title`, but the `elsif` is true because `site.description` does exist.

So our title becomes `page.title | site.description` :-1:

Comment (jhabdas) [inline lib/template.html]: Good eye, @pathawks. Resolved with https://github.com/jekyll/jekyll-seo-tag/pull/162/commits/51db5806c20f3947936a009afca8ae4a6c6230c2

File: lib/template.html [hunk 5]
```diff
@@ -5,21 +5,13 @@
...
-  {% assign seo_title = seo_site_title %}
-  {% assign seo_page_title = seo_site_title %}
-
-  {% if site.description %}
-    {% assign seo_title = seo_title | append:" - " | append: site.description %}
-  {% endif %}
+{% if page.title and seo_site_title %}
+  {% assign seo_title = seo_title | append:" | " | append: seo_site_title %}
+{% elsif site.description and seo_site_title %}
+  {% assign seo_title = seo_title | append:" | " | append: site.description %}
```
Comment (pathawks) [inline lib/template.html]: Can we use `seo_site_title` here, just to make explicit what we are doing?

```
{% assign seo_title = seo_site_title | append:" | " | append: site.description %}
```

We could probably get rid of the assignment on L9 and also change l12 to refer to `page.title`

Since it is what we're testing for, it seems like that should be what we use.

Comment (jhabdas) [inline lib/template.html]: Agreed. Updated with https://github.com/jekyll/jekyll-seo-tag/pull/162/commits/2a48f272f68b309b108b4cda27a862e6744c1f20. Had to keep L9 as it contains some default logic which, if would remove, would break some of the tests introduced.

**Comment by pathawks** [comment]
I'm still pretty indifferent on the specific title separator, but the implementation looks good to me.

@benbalter?

**Comment by benbalter** [comment]
@jekyllbot: merge +minor.

**Comment by jhabdas** [comment]
My pleasure, and thanks for letting the refactoring in with the PR. It's pragmatic and helps address the refactoring I wanted to tackle for #161. I ran the updated test suite against the previous template logic after a replacing all instances of `\|` with `-` in the spec and we're 1:1 with previous functionality, now with better codecov.

**Comment by benbalter** [comment]
@jhabdas looks like it didn't actually merge. Can you undelete the branch so I can reopen the PR and merge manually?

**Comment by jhabdas** [comment]
@benbalter done

**Comment by nhaglind** [comment]
What's the deal with the change here? I have a number of sites that use `|` as a title separator `About Something | Company Name` and now after the latest update that has been removed in serving and the build and becomes `About Something Company Name`. Even if it's specifically included in front matter.

**Comment by benbalter** [comment]
> Even if it's specifically included in front matter.

@nhaglind you put both the page title and the site title in the front matter? Can you provide a link or code snippet to how you're using this? I don't see how it would affect that behavior.

**Comment by jhabdas** [comment]
@nhaglind titles are optional with this plugin. If you were overriding to achieve the pipe separator you may already be customized. Please check release notes and README for usage, and file a bug if you think you've found an edge case.

**Comment by nhaglind** [comment]
Yes, absolutely. 

```
---
permalink: /foobar-noodles
title: Foobar Noodles | ACME Corp
description: Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce sit amet dui diam.
layout: default
image: /images/noodles.jpg
tags: ['foo', 'bar',]
---
```

This previously returned the title `Foobar Noodles | ACME Corp` and now returns `Foobar Noodles ACME Corp`.

**Comment by nhaglind** [comment]
A simple example of the behavior that should be reproducible using default of `{% seo %}`. 

`title: Foobar Noodles | ACME Corp` = `<title>Foobar Noodles ACME Corp</title>`
`title: Foobar Noodles - ACME Corp` = `<title>Foobar Noodles - ACME Corp</title>`

**Comment by benbalter** [comment]
@nhaglind does this happen in 2.2.0 or in 2.2.1 (just released moments ago) or both? Also, what Markdown engine are you using (I'm wondering if it's treating it as markdown).

**Comment by nhaglind** [comment]
@benbalter using latest version of jekyll 3.4.3 with default kramdown. This is related to 2.2.0 I'll update to 2.2.1 right now to see if this changes behavior.

**Comment by benbalter** [comment]
I was able to recreate this locally. The issue is that `"Foo | Bar" | markdownify`, at least with Kramdown, results in `Foo Bar`. While it's not the plugin's intention for users to put both the page title and site title in the page's `title` field, you can get around the issue by escaping the pipe, e.g., `Foo \| Bar`, which testing locally, should work.

**Comment by nhaglind** [comment]
You're correct escaping fixes `title` and `og` but it for Twitter it will include the escaping character. `<meta name="twitter:title" content="Foobar Noodles \| ACME Corp">`.

What would properly format all these things together? I have no problem doing an adjustment like the following but this update resulted in a sitewide change. 

`site: ACME Corp` in `_config.yml` then in each page `title: Foobar Noodles`?

Looking to achieve what was previously being returned prior to update.

**Comment by benbalter** [comment]
> `site: ACME Corp` in `_config.yml` then in each page `title: Foobar Noodles`?

Correct. That's the intended use (except: `title` in `_config.yml` instead of `site`). 

I believe with 2.2.1, they should be consistently escaped (as they use the same `markdownify` filter.

**Comment by nhaglind** [comment]
Looks like I got a lot of pages to update! But thanks a lot @benbalter for taking the time to look into this behavioral change.

**Comment by jhabdas** [comment]
@nhaglind You may want to consider updating your titles now to make use of the `|` syntax afforded by this update (since it seems like that's what you were trying to achieve). Please see the following resource for best practices on page titles: https://moz.com/learn/seo/title-tag

**Comment by cup** [comment]
https://github.com/jekyll/jekyll-seo-tag/issues/121

**Comment by cup** [comment]
https://github.com/jekyll/jekyll-seo-tag/issues/121
https://github.com/jekyll/jekyll-seo-tag/issues/331
https://github.com/jekyll/jekyll-seo-tag/pull/162
